tools/python/convert_onnx_model_to_ort.py

```diff
@@ -0,0 +1,56 @@
+import argparse
+import onnxruntime as ort
+import os
+import re
+
+
+def convert(model: str):
+
+    if not model.endswith('.onnx'):
+        raise ValueError("Model filename must end in .onnx.")
+
+    onnx_target_path = re.sub('.onnx$', '.optimized.onnx', model)
+    ort_target_path = re.sub('.onnx$', '.ort', model)
+
+    so = ort.SessionOptions()
+    so.optimized_model_filepath = onnx_target_path
+    so.add_session_config_entry('session.save_model_format', 'ONNX')
+    so.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_EXTENDED  # Skip NCHWc optimizations
+
+    print("Optimizing ONNX model {} and saving in ONNX format to {}".format(model, onnx_target_path))
+    # creating the session will result in the optimized model being saved
+    _ = ort.InferenceSession(model, sess_options=so)
+
+    # Second, convert optimized ONNX model to ORT format
+    so.optimized_model_filepath = ort_target_path
+    so.add_session_config_entry('session.save_model_format', 'ORT')
+
+    print("Converting optimized ONNX model {} to ORT format model {}".format(onnx_target_path, ort_target_path))
+    _ = ort.InferenceSession(onnx_target_path, sess_options=so)
+
+    orig_size = os.path.getsize(onnx_target_path)
+    new_size = os.path.getsize(ort_target_path)
+    print("Serialized {} to {}. Sizes: orig={} new={} Diff={} Ratio={:.4f}%".format(
```

Comment:
>Ratio={:.4f}% [](start = 62, length = 13)

This will output 150% as 1.5%, use {:4%}?

Reply:
Changed to output 'new:old={:.4f}:1.0'

---
In reply to: [483306631](https://github.com/microsoft/onnxruntime/pull/5056#discussion_r483306631) [](ancestors = 483306631)